spark/src/main/java/org/apache/iceberg/actions/BaseSparkAction.java

```diff
@@ -86,6 +126,32 @@
     return manifestDF.union(otherMetadataFileDF).union(manifestListDF);
   }
 
+  protected static Dataset<Row> loadMetadataTable(SparkSession spark, String tableName, String tableLocation,
+                                                  MetadataTableType type) {
+    DataFrameReader noCatalogReader = spark.read().format("iceberg");
+    if (tableName.contains("/")) {
+      // Metadata location passed, load without a catalog
```

Comment:
nit: It may not necessarily be a metadata location. It can be a valid Hadoop table location too.